### PR TITLE
feat(ci): Adds chainloop to codeql pipeline

### LINKED ISSUE #707: attests codeql process
see if we can also store the output, i.e sarif file.

Only run Chainloop if the current run is not provoked by a pull request.

### PR DESCRIPTION
**Pull request by javirln**
This PR adds chainloop CLI to codeql workflows to attest SARIF results.

Closes #707

### CODE REVIEW

**Review by migmartri — APPROVED** [review]
LGTM, but let's wait until we stabilize the other two jobs :)

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -48,3 +61,28 @@ jobs:
         uses: github/codeql-action/analyze@6a28655e3dcb49cb0840ea372fd6d17733edd8a4 # v2.21.8
         with:
           category: "/language:${{matrix.language}}"
+        outputs:
+          sarif-output: /tmp/codeql-results.sarif
```
Comment (migmartri) [inline .github/workflows/codeql.yml:65]: have you tried this? 

I am wondering if what the action documents is actually an output of the action, that can be then accessed with something like ` ${{ steps.analyze.outputs.sarif-output}}`

https://github.com/github/codeql-action/blob/4ebadbc7468649cf79b138f45e20d999351f6ed0/analyze/action.yml#L81

**Review by migmartri — CHANGES_REQUESTED** [review]
asking for changes to make sure we double check the output usage.

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (javirln) [reply .github/workflows/codeql.yml]: Added, the only problem is that it does not include the filename. I've added the changes so it uses the default one along with the absolute path take from the action.

**Comment by javirln** [comment]
Results being uploaded:
Screenshot 2024-04-29 at 10 38 20

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -45,6 +58,30 @@ jobs:
...
 
       - name: Perform CodeQL Analysis
+        id: codeqlresults
         uses: github/codeql-action/analyze@6a28655e3dcb49cb0840ea372fd6d17733edd8a4 # v2.21.8
         with:
           category: "/language:${{matrix.language}}"
+
+      - name: Add Attestation (Sarif results)
+        run: |
+          chainloop attestation add --name sarif-results --value ${{steps.codeqlresults.outputs.sarif-output}}/go.sarif
```
Comment (migmartri) [inline .github/workflows/codeql.yml:68]: nice :)

**Review by migmartri — APPROVED** [review]

**Pull request merged**